### PR TITLE
update boltdb lock

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @c4pt0r

File: store/localstore/kv_test.go [hunk 1]
```diff
@@ -555,3 +557,39 @@ func (s *testKVSuite) TestDBClose(c *C) {
...
+			txn.Set([]byte("b"), []byte("0"))
+			txn.Inc([]byte("b"), 1)
+
+			return nil
+		})
+
+		return nil
+	})
+
+	kv.RunInNewTxn(store, false, func(txn kv.Transaction) error {
```
Comment (zimulala) [inline store/localstore/kv_test.go]: The following test can be removed ?

Comment (siddontang) [inline store/localstore/kv_test.go]: no, need to check above transaction success.

Comment (zimulala) [inline store/localstore/kv_test.go]: Gotcha.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**